org.code.toboggan.filesystem/src/org/code/toboggan/filesystem/extensions/project/FSProjectCreate.java

```diff
@@ -0,0 +1,144 @@
+	private AbstractExtensionManager extMgr;
+	
+	public FSProjectCreate() {
+		this.ss = CoreActivator.getSessionStorage();
+		pc = new ProjectController(ss);
+		this.extMgr = FileSystemExtensionManager.getInstance();
+	}
+	
+	@Override
+	public void projectFetched(Project p) {
+		IWorkspace workspace = ResourcesPlugin.getWorkspace();
+		IProject iProject = workspace.getRoot().getProject(p.getName());
+
+		pc.createProject(p);
+		pc.putProjectLocation(iProject.getLocation().toFile().toPath(), p.getProjectID());
+		
+		Display.getDefault().syncExec(() -> PlatformUI.getWorkbench().saveAllEditors(false));	
+		CCIgnore ignoreFile = CCIgnore.createForProject(iProject);
+		
+		List<IFile> ifiles = recursivelyGetFiles(iProject, ignoreFile);
+		for (IFile f : ifiles) {
+			try (InputStream in = f.getContents();) {
+				String contents = new String(FSUtils.inputStreamToByteArray(in));
+				if (contents.contains("\r\n")) {
+					contents = contents.replace("\r\n", "\n");
+				}
+				Path fileLocation = f.getLocation().toFile().toPath();
+				new Thread(APIFactory.createFileCreate(f.getName(), fileLocation, p.getProjectID(), contents.getBytes())).start();
+			} catch (IOException | CoreException e) {
+				logger.error("Error reading files as part of project create", e);
+				new Thread(APIFactory.createFileDelete(p.getProjectID())).start();
+				return;
+			}			
+		}
+		
+		Set<ICoreExtension> extensions = extMgr.getExtensions(APIExtensionIDs.PROJECT_CREATE_ID);
+		for (ICoreExtension e : extensions) {
+			IFSProjectCreateExt createExt = (IFSProjectCreateExt) e;
+			createExt.projectCreated(p, iProject);
+		}
+	}
+	
+	private List<IFile> recursivelyGetFiles(IContainer f, CCIgnore ignoreFile) {
+		List<IFile> files = new ArrayList<>();
+		IResource[] members = null;
+		
+		try {
+			members = f.members();
+		} catch (CoreException e1) {
+			e1.printStackTrace();
+		}
+		
+		for(IResource m : members) {
+			if (m instanceof IFile) {
+				String path = ((IFile) m).getProjectRelativePath().toString();
+				if (ignoreFile.containsEntry(path)) {
+					logger.debug(String.format("File %s was ignored when scanning for files.", path));
+				} else {
+					files.add((IFile) m);
+				}
+			} else if (m instanceof IFolder) {
+				String path = ((IFolder) m).getProjectRelativePath().toString();
+				if (ignoreFile.containsEntry(path)) {
+					logger.debug(String.format("Folder %s was ignored when scanning for files.", path));
+				} else {
+					files.addAll(recursivelyGetFiles((IFolder) m, ignoreFile));
+				}
+			}
+		}
+		return files;
+	}
+
+	@Override
+	public void projectFetchFailed(long projectId) {
+		// TODO Auto-generated method stub
+		
+	}
+	
+	@Override
+	public void projectCreated(long projectId) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public void projectCreationFailed(String name) {
+		// TODO Auto-generated method stub
+		
+	}
+
+	@Override
+	public void subscribed(long projectId) {
+		// TODO Auto-generated method stub
```

Comment:
remove the `TODO Auto-generated` comment and replace it with a `// Do nothing` if we truely don't need to implement these functions